### PR TITLE
minor refactors and improvements to the Http Rest layer

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Omit
Docs: Omit

This was done when starting to research https://github.com/conan-io/conan/issues/6240

- Removing dead code
- Minor pep8 and linting
- Moving _post_json to V1 only
- Removing unnecessary inheritance in Router

Seems there was a weird ``'Accept': 'text/plain'`` which was not being used, removed it.

#tags: slow
#revisions: 1

### CODE REVIEW

File: conans/client/rest/rest_client_common.py [hunk 1]
```diff
@@ -175,7 +169,6 @@ def get_json(self, url, data=None):
         headers = self.custom_headers
         if data:  # POST request
             headers.update({'Content-type': 'application/json',
-                            'Accept': 'text/plain',
```
Comment (memsharded) [inline conans/client/rest/rest_client_common.py]: This was very weird, it seems the `text/plain` was just being overwritten by json (later in declaration)

File: conans/client/rest/rest_client_common.py [hunk 2]
```diff
@@ -22,24 +22,19 @@ def __call__(self, request):
...
-        return None
+    tmp = {v: k for k, v in EXCEPTION_CODE_MAPPING.items()  # All except NotFound
+           if k not in (RecipeNotFoundException, PackageNotFoundException)}
+    if error_code in tmp:
+        logger.debug("REST ERROR: %s" % str(tmp[error_code]))
+        return tmp[error_code]
+    else:
+        base_error = int(str(error_code)[0] + "00")
+        logger.debug("REST ERROR: %s" % str(base_error))
+        try:
```
Comment (danimtb) [inline conans/client/rest/rest_client_common.py]: This should cover both return lines to catch any exiting KeyError

File: conans/client/rest/rest_client_v1.py [hunk 3]
```diff
@@ -308,3 +309,12 @@ def get_latest_recipe_revision(self, ref):
 
     def get_latest_package_revision(self, pref):
         raise NoRestV2Available("The remote doesn't support revisions")
+
+    def _post_json(self, url, payload):
```
Comment (danimtb) [inline conans/client/rest/rest_client_v1.py]: is there no `_post_json` call in V2?

Comment (danimtb) [reply conans/client/rest/rest_client_v1.py]: Having a look at the code I wonder if this is needed. I think a `get_json()` with `data` argument should be the same. the only difference I see are the custom_headers

Comment (memsharded) [reply conans/client/rest/rest_client_v1.py]: No, it was not being used in V2, this is why I moved it to the place it is used (reducing unnecessary big scope).

The problem is that ``get_json()`` does more things: add headers, interprets return codes and raise exceptions. That wouldn't be a refactor, but probably change behavior, I don't want to do that here.

File: conans/client/rest/rest_client_common.py (hunk 2 above)
Comment (memsharded) [reply conans/client/rest/rest_client_common.py]: All potential Key errors are covered here.  The above is conditioned to ``if error_code in tmp``

File: conans/util/tracer.py [hunk 4]
```diff
@@ -138,8 +138,10 @@ def log_package_built(pref, duration, log_run=None):
 
 def log_client_rest_api_call(url, method, duration, headers):
     headers = copy.copy(headers)
-    headers["Authorization"] = MASKED_FIELD
-    headers["X-Client-Anonymous-Id"] = MASKED_FIELD
+    if "Authorization" in headers:
```
Comment (danimtb) [inline conans/util/tracer.py]: 👏 This is what I was talking about the other day 👍

**Pull request merged**